### PR TITLE
Get rid of ancient, unused option to track unique reads in a LIBS

### LINKED ISSUE #1495: remove mentions of SAMRecordAlignmentState
SAMRecordAlignmentState is some ancient name that does not exist even in GATK3. We should purge it from the comments

### PR DESCRIPTION
**Pull request by davidbenjamin**
Closes #1495.  @droazen Adam's comments suggest that this feature has been unused since before GATK 3.  May we put it to rest?

### CODE REVIEW

**Comment by samuelklee** [comment]
@davidbenjamin happy to give this a look and a review after a rebase!

**Comment by davidbenjamin** [comment]
@samuelklee I would be thrilled to rebase and hand it to you for review!

**Comment by davidbenjamin** [comment]
@samuelklee It's rebased and passing tests again.

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java [hunk 1]
```diff
@@ -124,30 +116,25 @@ public LocusIteratorByState(final Iterator<GATKRead> samIterator,
 
     /**
      * Create a new LocusIteratorByState
-     *
-     * @param samIterator                    the iterator of reads to process into pileups.  Reads must be ordered
+     *  @param samIterator                    the iterator of reads to process into pileups.  Reads must be ordered
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:119]: stray whitespace introduced before `@param`

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java [hunk 2]
```diff
@@ -124,30 +116,25 @@ public LocusIteratorByState(final Iterator<GATKRead> samIterator,
...
      *                                       according to standard coordinate-sorted BAM conventions
      * @param downsamplingMethod             information about how to downsample the reads
-     * @param keepUniqueReadListInLIBS       if true, we will keep the unique reads from the samIterator and make them
-     *                                       available via the transferReadsFromAllPreviousPileups interface
      * @param samples                        a complete list of samples present in the read groups for the reads coming from samIterator.
-     *                                       This is generally just the set of read group sample fields in the SAMFileHeader.  This
-     *                                       list of samples may contain a null element, and all reads without read groups will
-     *                                       be mapped to this null sample
+ *                                       This is generally just the set of read group sample fields in the SAMFileHeader.  This
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:123]: some more whitespace to fix up here

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java [hunk 3]
```diff
@@ -188,23 +171,19 @@ public LocusIteratorByState(final Iterator<GATKRead> samIterator,
 
     /**
      * Create a new LocusIteratorByState
-     *
-     * @param samIterator                    the iterator of reads to process into pileups.  Reads must be ordered
+     *  @param samIterator                    the iterator of reads to process into pileups.  Reads must be ordered
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:174]: same deal here and below

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java [hunk 4]
```diff
@@ -366,40 +345,4 @@ private boolean dontIncludeReadInPileup(final GATKRead rec, final long pos) {
         return ReadUtils.isBaseInsideAdaptor(rec, pos);
     }
 
-    /**
```
Comment (samuelklee) [inline src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:369]: Not in this PR, but `dontIncludeReadInPileup` can be made static. IntelliJ also says that we can get rid of the `samples` field; just pass a new copy to the `ReadStateManager` in line 205.

**Review by samuelklee — APPROVED** [review]
@davidbenjamin just a few minor fixups, otherwise LGTM! Although I might expand the PR description to mention that the option was unused in production code, but you also removed some test code in which it was used. And perhaps ping @droazen for a final check.

Otherwise, thanks for cleaning this up. Here's to deleting (doubly) ancient code!

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java (hunk 2 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:123]: done

File: src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java (hunk 4 above)
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/utils/locusiterator/LocusIteratorByState.java:369]: done and done

**Pull request merged**